modules/layers/src/column-layer/column-geometry.js

```diff
@@ -40,18 +47,20 @@ function tesselateColumn(props) {
   //
   for (let j = 0; j < vertsAroundEdge; j++) {
     const a = j * stepAngle;
-    const vertex = vertices && vertices[j % nradial];
-    const nextVertex = vertices && vertices[(j + 1) % nradial];
+    const vertexIndex = j % nradial;
+    const nextVertexIndex = (j + 1) % nradial;
     const sin = Math.sin(a);
     const cos = Math.cos(a);
 
     for (let k = 0; k < 2; k++) {
-      positions[i + 0] = vertex ? vertex[0] : cos * radius;
-      positions[i + 1] = vertex ? vertex[1] : sin * radius;
+      positions[i + 0] = vertices ? vertices[vertexIndex * 2] : cos * radius;
+      positions[i + 1] = vertices ? vertices[vertexIndex * 2 + 1] : sin * radius;
       positions[i + 2] = (1 / 2 - k) * height;
 
-      normals[i + 0] = vertex ? nextVertex[0] - vertex[0] : cos;
-      normals[i + 1] = vertex ? nextVertex[1] - vertex[1] : sin;
+      normals[i + 0] = vertices ? vertices[vertexIndex * 2] - vertices[nextVertexIndex * 2] : cos;
```

Comment:
currentVertex - nextVertex doesn't look like the direction of normal here. Is it intentional? For same config N(cos, sin) !== N(currentVertex - nextVertex)

Reply:
Hmmm you are right, that seems to be another bug in this class. It should just be `currentVertex` (we are not doing flat shading in this layer)